packages/message-manager/src/EncryptionPublicKeyManager.ts

```diff
@@ -0,0 +1,149 @@
+import { v1 as random } from 'uuid';
+import {
+  AbstractMessageManager,
+  AbstractMessage,
+  AbstractMessageParams,
+  AbstractMessageParamsMetamask,
+  OriginalRequest,
+} from './AbstractMessageManager';
+import { validateEncryptionPublicKeyMessageData } from './utils';
+
+/**
+ * @type EncryptionPublicKey
+ *
+ * Represents and contains data about a 'eth_getEncryptionPublicKey' type request.
+ * These are created when an encryption public key is requested.
+ * @property id - An id to track and identify the message object
+ * @property messageParams - The parameters to pass to the eth_getEncryptionPublicKey method once the request is approved
+ * @property type - The json-prc method for which an encryption public key request has been made.
+ * A 'Message' which always has a 'eth_getEncryptionPublicKey' type
+ * @property rawSig - Encryption public key
+ */
+export interface EncryptionPublicKey extends AbstractMessage {
+  messageParams: EncryptionPublicKeyParams;
+}
+
+/**
+ * @type EncryptionPublicKeyParams
+ *
+ * Represents the parameters to pass to the method once the request is approved.
+ * @property from - Address from which to extract the encryption public key
+ * @property origin? - Added for request origin identification
+ */
+export type EncryptionPublicKeyParams = AbstractMessageParams;
```

Comment:
Yet another hacky behaviour. Since the parameter is an address and there's no additional data, I'm using `AbstractMessageParams.from`, which is an ethereum address.

Reply:
I actually quite like this, it's open to extension (though unlikely as deprecated) and matches the other message managers. But it makes conceptual sense too, as it's where the encrypted data came "from".

Perhaps worth changing the comment in the original `AbstractMessageParams` type though for clarity?

The whole thing is really `AbstractSigningMessageManager` but we can gently stretch it's scope a bit.